src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,286 @@
+        gasUsage[0] = 1000000;
+        gasUsage[1] = 1000000;
+        minGasPerMinute[0] = 700;
+        minGasPerMinute[1] = 700;
+
+        SUBSIDY.setGasUsageAndMinGasPerMinute(criterias, gasUsage, minGasPerMinute);
+    }
+
+    /**
+     * @notice Computes the criteria that is passed when claiming subsidy.
+     * @param _inputAssetA The input asset
+     * @param _outputAssetA The output asset
+     * @return The criteria
+     */
+    function computeCriteria(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint64
+    ) public pure override (BridgeBase) returns (uint256) {
+        return _computeCriteria(_inputAssetA.erc20Address, _outputAssetA.erc20Address);
+    }
+
+    /**
+     * @notice Deposits Curve LP tokens
+     * @dev RCT is minted for the bridge. Mirrors balance of minted Convex LP tokens for the bridge.
+     * @param _outputAssetA Asset for the RCT token
+     * @param _totalInputValue Number of Curve LP tokens to stake
+     * @param _selectedPool Pool info about the staking pool
+     * @return outputValueA Number of successfully staked Curve LP tokens
+     */
+    function _deposit(
+        AztecTypes.AztecAsset memory _outputAssetA,
+        uint256 _totalInputValue,
+        PoolInfo memory _selectedPool
+    ) internal returns (uint256 outputValueA) {
+        uint256 startCurveRewards = ICurveRewards(_selectedPool.curveRewards).balanceOf(address(this));
+
+        BOOSTER.deposit(_selectedPool.poolId, _totalInputValue, true);
+
+        uint256 endCurveRewards = ICurveRewards(_selectedPool.curveRewards).balanceOf(address(this));
+
+        outputValueA = (endCurveRewards - startCurveRewards);
+
+        IRepConvexToken(_outputAssetA.erc20Address).mint(_totalInputValue);
+    }
+
+    /**
+     * @notice Withdraws Curve LP tokens.
+     * @dev RCT is burned for the bridge. Mirrors balance of minted Convex LP tokens for the bridge.
+     * @param _inputAssetA Asset for the RCT token.
+     * @param _outputAssetA Asset for the Curve LP token.
+     * @param _totalInputValue Number of Curve LP tokens to unstake.
+     * @return outputValueA Number of withdrawn Curve LP tokens.
+     */
+    function _withdraw(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        uint256 _totalInputValue,
+        uint256 _interactionNonce
+    ) internal returns (uint256 outputValueA, uint256 outputValueB) {
+        PoolInfo memory selectedPool = pools[_outputAssetA.erc20Address];
+
+        uint256 startCurveLpTokens = IERC20(_outputAssetA.erc20Address).balanceOf(address(this));
+
+        // transfer ConvexLP tokens from CrvRewards back to the bridge
+        ICurveRewards(selectedPool.curveRewards).withdraw(_totalInputValue, true); // always claim rewards
```

Comment:
This claims the rewards for the whole bridge, right? If that is the case then there is a bit of an issue with to whom you are returning the rewards. You are essentially returning the rewards of all the users to the users of the current round. If that is the case then it's a tricky situation and accounting might get complicated.

Reply:
Correct, it claims rewards for the whole bridge each time the withdraw function is called. Should we introduce some accounting management and proportional pay off? This would most likely turn the bridge into a stateful bridge

Reply:
Yes, the user should receive only his rewards and not the rewards of the whole bridge.